File: src/main/java/ee/upcourse/trainingmanager/model/Topic.java
package ee.upcourse.trainingmanager.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "topic")
public class Topic {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    @ManyToOne(cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    @JoinColumn(name = "course_id")
    private Course course;

    @OneToMany(mappedBy = "topic",
    cascade = CascadeType.ALL)
    private List<Lesson> lessons;

}

File: angularclient/src/app/common/topic.ts
import {Lesson} from "./lesson";

export class Topic {

  id: number;
  name: string;
  course: string;
  lessons: Lesson[];
  private _teacher: string;

  constructor(name: string, course: string) {
    this.name = name;
    this.course = course;
  }


  set teacher(value: string) {
    this._teacher = value;
  }
}
